src/framework/preferences/js/CaptionsEnactor.js

```diff
@@ -0,0 +1,245 @@
+    fluid.prefs.enactor.captions.waitForYouTubeAPI = function (that) {
+        var promise = fluid.promise();
+        promise.then(function () {
+            that.locate("videos").each(function (index, elm) {
+                that.events.onVideoElmLocated.fire($(elm));
+            });
+        }, function (error) {
+            fluid.log(fluid.logLevel.WARN, error);
+        });
+
+        if (fluid.get(window, ["YT", "Player"])) {
+            promise.resolve();
+        } else {
+            // the YouTube iframe api will call onYouTubeIframeAPIReady after the api has loaded
+            window.onYouTubeIframeAPIReady = promise.resolve;
+        }
+
+        return promise;
+    };
+
+    /**
+     * See: https://developers.google.com/youtube/iframe_api_reference#Events for details on the YouTube player
+     * events. This includes when they are fired and what data is passed along.
+     */
+    fluid.defaults("fluid.prefs.enactor.captions.youTubePlayer", {
+        gradeNames: ["fluid.viewComponent"],
+        events: {
+            onReady: null,
+            onStateChange: null,
+            onPlaybackQualityChange: null,
+            onPlaybackRateChange: null,
+            onError: null,
+            onApiChange: null
+        },
+        model: {
+            captions: false,
+            track: {}
+        },
+        members: {
+            player: {
+                expander: {
+                    funcName: "fluid.prefs.enactor.captions.youTubePlayer.initYTPlayer",
+                    args: ["{that}"]
+                }
+            },
+            tracklist: []
+        },
+        invokers: {
+            applyCaptions: {
+                funcName: "fluid.prefs.enactor.captions.youTubePlayer.applyCaptions",
+                args: ["{that}.player", "{that}.model.track", "{that}.model.captions"]
+            }
+        },
+        modelListeners: {
+            "setCaptions": {
+                listener: "{that}.applyCaptions",
+                path: ["captions", "track"],
+                excludeSource: "init"
+            }
+        },
+        listeners: {
+            "onApiChange.prepTrack": {
+                listener: "fluid.prefs.enactor.captions.youTubePlayer.prepTrack",
+                args: ["{that}", "{that}.player"]
+            },
+            "onApiChange.applyCaptions": {
+                listener: "{that}.applyCaptions",
+                priority: "after:prepTrack"
+            }
+        }
+    });
+
+    /**
+     * Convert queryString into a parameter object. Note that repeated parameters will be replaced with the last one.
+     *
+     * @param {String} queryStr - the query string to parse. It assumes the first character is not "?";
+     *
+     * @return {Object} - An Object representation of the key/value pairs from the query string.
+     */
+    fluid.prefs.enactor.captions.youTubePlayer.parseQueryString = function (queryStr) {
```

Comment:
I believe this function is not required in the modern world - https://developer.mozilla.org/en-US/docs/Web/API/URL

Reply:
It doesn't look like the URL API is supported in IE 11 ( https://caniuse.com/#search=URL ). I'll leave a comment that this is deprecated and should be switched to use the URL API after IE 11 is no longer supported.